algorithms/kernel/k_nearest_neighbors/kdtree_knn_impl.i

```diff
@@ -160,11 +139,27 @@ public:
     }
 
 private:
-    T * _data;
-    size_t _top;
-    size_t _count;
-    size_t _size;
-    size_t _sizeMinus1;
+    T * _data          = nullptr;
+    size_t _top        = 0;
+    size_t _sizeMinus1 = 0;
+    size_t _size       = 0;
+    DAAL_FORCEINLINE void setSize(size_t size)
+    {
+        _size       = size;
+        _sizeMinus1 = static_cast<size_t>(_size - 1);
```

Comment:
I think static_cast is not necessary here.

Reply:
But it doesn't affect any functionality or even performance

Reply:
It affects understanding the code, If you use extra constructions there :)
I just spend 0.5 minutes to understand reason why you used the cast and didn't find).
So, please, save 0.5 minutes of other people who will read such lines of code (not only here, but in the future in general).

Reply:
I agree with Egor. Remove static_cast.